lib/apartment/adapters/abstract_adapter.rb

```diff
@@ -224,6 +231,34 @@ def rescuable_exceptions
       def rescue_from
         []
       end
+
+      def extended_cloned_config(tenant)
+        @config.clone.with_indifferent_access.tap do |config|
+          config.merge!(Apartment.db_config_for(tenant))
```

Comment:
> whenever you need to work on another server, it spares a few keys in the specific config (basically adapter, pooling conf etc...)

but won't this info only ever be provided in config, and usually by some database lookup or proc? (and if so i don't really see the convenience as being a thing)

> whenever you want to keep a tenant on the same server, you just pass an empty hash

same server as what? it'll be the same server as you were last connected to, which seems unpredictable and error prone to me ..